modules/shortcodes/youtube.php

```diff
@@ -365,16 +294,162 @@ function youtube_id( $url ) {
 	return $html;
 }
 
+/**
+ * Gets the query args present in the YouTube shortcode URL.
+ *
+ * @since 8.0.0
+ *
+ * @param string $url The URL of the shortcode.
+ *
+ * @return array|false The query args of the URL, or false.
+ */
+function jetpack_shortcode_youtube_query_args( $url ) {
+	$qargs = array();
+	if ( ! empty( $url['query'] ) ) {
+		wp_parse_str( $url['query'], $qargs );
+	} else {
+		return false;
+	}
+
+	$fargs = array();
+	if ( ! empty( $url['fragment'] ) ) {
+		wp_parse_str( $url['fragment'], $fargs );
+	}
+
+	return array_merge( $fargs, $qargs );
+}
+
 /**
  * Display the Youtube shortcode.
  *
  * @param array $atts Shortcode attributes.
+ *
+ * @return string The rendered shortcode.
  */
 function youtube_shortcode( $atts ) {
-	return youtube_id( ( isset( $atts[0] ) ) ? ltrim( $atts[0], '=' ) : shortcode_new_to_old_params( $atts ) );
+	if (
+		class_exists( 'Jetpack_AMP_Support' )
+		&& Jetpack_AMP_Support::is_amp_request()
+	) {
+		return jetpack_amp_youtube_shortcode( $atts );
+	} else {
+		return youtube_id( ( isset( $atts[0] ) ) ? ltrim( $atts[0], '=' ) : shortcode_new_to_old_params( $atts ) );
```

Comment:
What do you think about setting the URL at the beginning of the `youtube_shortcode` function, so you don't have to repeat that `isset( $atts[0] ) ) ? ltrim( $atts[0], '=' ) : shortcode_new_to_old_params( $atts )` twice (here and in the `jetpack_amp_youtube_shortcode` function)?

Reply:
That's a good idea. Is https://github.com/Automattic/jetpack/pull/14010/commits/51a5eb78c534d1c9b7c2edef2718b04d167fb5f4 what you had in mind?

Reply:
That's a good idea, it should be more DRY. How does https://github.com/Automattic/jetpack/pull/14010/commits/51a5eb78c534d1c9b7c2edef2718b04d167fb5f4 look?